### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by dheid**

### CODE REVIEW

File: src/main/java/net/anotheria/anoprise/dataspace/DataspaceServiceImpl.java [hunk 1]
```diff
@@ -51,7 +51,7 @@ public DataspaceServiceImpl() {
 		try {
 			cache = Caches.createConfigurableHardwiredCache("ano-prise-dataspace-cache");
 		} catch (IllegalArgumentException e) {
-			log.warn("Can't find cache configuration for ano-prise-dataspace-cache, falling back to min cache.");
+			log.warn("Can't find cache configuration for ano-prise-dataspace-cache, falling back to min cache.", e);
```
Comment (dvayanu) [inline src/main/java/net/anotheria/anoprise/dataspace/DataspaceServiceImpl.java:54]: exception was ignored on purpose, the log message already says what happened, no need to further spam.

File: src/main/java/net/anotheria/anoprise/dataspace/persistence/DataspaceNotFoundException.java [hunk 2]
```diff
@@ -8,11 +8,6 @@
 public class DataspaceNotFoundException extends DataspacePersistenceServiceException {
 
 	/**
-	 * Basic serialVersionUID variable.
-	 */
-	private static final long serialVersionUID = 4675725133571302930L;
-
-	/**
 	 * Public constructor.
```
Comment (dvayanu) [inline src/main/java/net/anotheria/anoprise/dataspace/persistence/DataspaceNotFoundException.java:11]: Why did you remove serialversionuid?

**Comment by dvayanu** [comment]
Hi Daniel,

once again we value your effort. But this pull request doesn't contain anything except for style changes and style is discussable, but not absolute.

In particular: 
- I really dislike usage of synchronized(this) as first statement in the method instead of synchronized keyword on the method. What is the expected benefit of this? 
- Why would you remove serialVersionUID from Exception classes?
- You use only snapshot versions again. There is really no need to have the projects to depend on latest versions only. If we would do this, we would need to release EVERYTHING on every lib change. Instead its the task of the actual end-artifact project to define versions. The dependencies in the ano-\* projects are set to the required minimum. However, the upgrade of configureme version in ano-prise doesn't make sense, as long as ano-prise doesn't use the new features.
- Pull request size. It is hard to approve 1000 lines +/- at once.

File: src/main/java/net/anotheria/anoprise/dataspace/DataspaceServiceImpl.java (hunk 1 above)
Comment (dheid) [inline src/main/java/net/anotheria/anoprise/dataspace/DataspaceServiceImpl.java]: It’s  better to use log levels instead of hiding possible important debugging information.

File: src/main/java/net/anotheria/anoprise/dataspace/persistence/DataspaceNotFoundException.java (hunk 2 above)
Comment (dheid) [inline src/main/java/net/anotheria/anoprise/dataspace/persistence/DataspaceNotFoundException.java]: Because this is exception is not serialized and deserialized anywhere.

**Comment by dheid** [comment]
Hi Leon,

There’s no need to repeat yourself. I got the point and already reacted many times to your comments. I won’t repeat, what I’ve already explained.

The benefit of using SNAPSHOT-Versions in POMs is to check the compatibility to recent changes. Since a new version also contains bug fixes and some deprecated methods and classes could be removed, it is very meaningful to use snapshot versions. Thus you need to release these versions first and then change them in the dependencies of related projects. It’s simply about the order of releasing.

Thanks for your patience.

Regards,

Daniel

File: src/main/java/net/anotheria/anoprise/dataspace/persistence/DataspaceNotFoundException.java (hunk 2 above)
Comment (dvayanu) [inline src/main/java/net/anotheria/anoprise/dataspace/persistence/DataspaceNotFoundException.java:11]: > Because this is exception is not serialized and deserialized anywhere.

How do you know? Have you checked all projects that are using this service? 
I would even say that your company has used this service as distributed service some time ago, and therefore has serialized / deserialized this exact exception...
